mogp_emulator/MultiOutputGP.py

```diff
@@ -170,28 +218,155 @@ def predict(self, testing, unc=True, deriv=True, include_nugget=True, processes=
+        obtain a list of emulators that have successfully been fit.
+
+        Returns a list of ``GaussianProcess`` objects which have been
+        fit (i.e. those which have a current valid set of
+        hyperparameters).
+
+        :returns: List of ``GaussianProcess`` objects indicating the
+                  emulators that have been fit. If no emulators
+                  have been fit, returns an empty list.
+        :rtype: list of ``GaussianProcess`` objects
+
+        """
+
+        return [gpem for (failed_fit, gpem)
+                in zip([em.theta is None for em in self.emulators],
+                       self.emulators) if not failed_fit]
+
+    def get_emulators_not_fit(self):
+        """Returns the indices of the emulators that have not been fit
+
+        When a ``MultiOutputGP`` class is initialized, none of the
+        emulators are fit. Fitting is done by passing the object to an
+        external fitting routine, which modifies the object to fit the
+        hyperparameters. Any emulators where the fitting fails are
+        returned to the initialized state, and this method is used to
+        obtain a list of emulators that have not been fit.
+
+        Returns a list of ``GaussianProcess`` objects which have
+        not been fit (i.e. those which do not have a current set of
+        hyperparameters).
+
+        :returns: List of ``GaussianProcess`` objects indicating the
+                  emulators that have not been fit. If all emulators
+                  have been fit, returns an empty list.
+        :rtype: list of ``GaussianProcess`` objects
 
-    def __str__(self):
         """
-        Returns a string representation of the model
 
-        :returns: A string representation of the model (indicates number of sub-components
-                  and array shapes)
-        :rtype: str
+        return [gpem for (failed_fit, gpem)
+                in zip([em.theta is None for em in self.emulators],
+                       self.emulators) if failed_fit]
+
+        
+    def __str__(self):
+        """Returns a string representation of the model
+
+        :returns: A string representation of the model (indicates
+                  number of sub-components and array shapes) :rtype:
+                  str
         """
 
         return ("Multi-Output Gaussian Process with:\n"+
                  str(self.n_emulators)+" emulators\n"+
                  str(self.n)+" training examples\n"+
                  str(self.D)+" input variables")
 
+
+def _gp_predict_default_NaN(gp, testing, unc, deriv, include_nugget):
+    """Wrapper function for the ``GaussianProcess`` predict method that
+    returns NaN if the GP is not fit. Allows ``MultiOutputGP`` objects
+    that do not have all emulators fit to still return predictions
+    with unfit emulator predictions replaced with NaN
+    """
+    
+    try:
+        return GaussianProcess.predict(gp, testing, unc, deriv,
```

Comment:
Is there a reason here not to just do ```return gp.predict(testing, unc, deriv, include_nugget)``` ?
(This way when we have the GPU branch, "gp" could be either a `GaussianProcess` or a `GaussianProcessGPU` instance)

Reply:
Yes, you are right -- I think I had the fact that`MultiOutputGP.predict` calls `GaussianProcess.predict` to take advantage of multiprocessing on my mind and forgot we might have other GP versions. I'll change prior to merging.